### DISCUSSION TITLE
TLS 1.3 testing endpoints

### DISCUSSION THREAD

**Discussion by sampov2** [aws/aws-sdk-java#2771] [Q&A] [answered]
Hi,

I just received information and warning regarding [older AWS Java SDK:s and TLS 1.3 support](https://aws.amazon.com/blogs/developer/tls-1-3-incompatibility-with-aws-sdk-for-java-versions-1-9-5-to-1-10-31/) coming this July. Are there new generation endpoints that could be used to validate whether applications can support the new TLS settings?

 Thanks

**Comment by debora-ito** [comment, marked as answer]
Hi @sampov2, some AWS services support TLS 1.3 currently. Note that these are not testing endpoints but actual service endpoints:

- KMS - according to the [KMS Dev Guide](https://docs.aws.amazon.com/kms/latest/developerguide/data-protection.html): "AWS KMS supports TLS 1.0—1.3 and hybrid post-quantum TLS for AWS KMS service endpoints in all regions, except AWS GovCloud (US) and China Regions."
- CloudFront - Amazon CloudFront announces support for TLSv1.3 for viewer connections - https://aws.amazon.com/about-aws/whats-new/2020/09/cloudfront-tlsv1-3-support/
